noxfile.py

```diff
@@ -10,28 +10,34 @@
 ).split(",")
 
 
-def deps(session):
+def deps(session, local_install):
     session.install("--upgrade", "setuptools", "pip")
-    session.install("-r", "requirements/dev.txt", ".", silent=True)
+    extra_flags = ["-e"] if local_install else []
+    session.install("-r", "requirements/dev.txt", *extra_flags, ".", silent=True)
 
 
 @nox.session(python=PYTHON_VERSIONS)
 def omegaconf(session):
-    deps(session)
+    deps(session, local_install=False)  # ensure we test the installed version
     session.run("pytest")
 
 
 @nox.session
 def docs(session):
-    deps(session)
+    deps(session, local_install=True)
     session.chdir("docs")
     session.run("sphinx-build", "-W", "-b", "doctest", "source", "build")
     session.run("sphinx-build", "-W", "-b", "html", "source", "build")
 
 
 @nox.session(python=PYTHON_VERSIONS)
 def coverage(session):
-    deps(session)
+    # For coverage, we must use the local installation because
```

Comment:
The problem is not whether or not we want to check grammar files for coverage, it's that the grammar files would be missing if we didn't use the local installation.

Reply:
Oh ok, there was a misunderstanding, I actually meant the *generated* grammar files, my bad.
The comment on this line is supposed to explain the issue, but maybe it's still not clear enough:

```
    # For coverage, we must use the editable installation because
    # `coverage run -m pytest` prepends `sys.path` with "." (the current
    # folder), so that the local code will be used in tests even if we set
    # `editable_installl=False`. This would cause problems due to potentially
    # missing the generated grammar files.
```

Trying to rephrase: if you install with `editable_install=False`, then `pip` will generate the grammar files in a folder that is not the current local folder, but when running `coverage run -m pytest` we are using the current local `omegaconf` folder because `.` is the first item in `sys.path` (and we can't really change that). As a result, we end up running tests without the generated grammar files => boom.

To see the issue, checkout this branch, edit `noxfile.py` so that `coverage` uses `editable_installl=False`, then run `python setup.py clean` and `nox -s coverage-3.8`